packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -448,18 +448,16 @@ export default class NewVm extends BaseComponent {
       : createVm(data)
   }
 
-  _initTemplate = template => {
-    if (!template) {
-      return this._reset()
-    }
-
-    this._setState({ template })
+  _onChangeTemplate = template =>
+    this._setState({ template }, () => this._initTemplate())
```

Comment:
`() => this._initTemplate()` → `this._initTemplate`

Reply:
Although this works, it prevents React from optimizing multiple synchronous `setState` calls. Isn't there a way to keep passing the template instead of relying on the state?